components/scream/src/share/grid/remap/abstract_remapper.hpp

```diff
@@ -178,9 +218,10 @@ class AbstractRemapper
   int           m_num_fields;
   int           m_num_registered_fields;
 
-  // Whether fields have been provided. Recall that the user may register
-  // fields passing only an identifier, and actually bind fields only
-  // at a later moment.
+  // This vector stores a boolean whose elements are the indices of fields that
+  // have been bound to the remapper. This vector is necessary because the
+  // binding of fields is separate from their regіstration: recall that one may
+  // register a field using its identifier, and bind the actual field later.
   std::vector<bool>   m_fields_are_bound;
```

Comment:
std::vector<bool> is a weird beast. I am not sure if we should keep it as is, or replace with vector<int>. As of now, all use cases of it do things correctly, but there's a risk someone might use this in derived classes, and do something dangerous like `bool* b = &m_fields_are_bound[i];`.

Perhaps we could add a WARNING, telling devs that they should be aware of vector<bool> peculiar behavior in memory.

Reply:
This is one of those areas where C++'s complexity makes all sorts of nastiness possible. I'll add a comment about it.